DevEdu.API/DevEdu.API/Controllers/CommentController.cs

```diff
@@ -12,42 +11,49 @@ namespace DevEdu.API.Controllers
     [Route("api/[controller]")]
     public class CommentController : Controller
     {
-        public CommentController()
+        private readonly ICommentRepository _commentRepository;
+        private readonly IMapper _mapper;
+        public CommentController(IMapper mapper, ICommentRepository commentRepository)
         {
+            _commentRepository = commentRepository;
+            _mapper = mapper;
         }
 
         //  api/comment/5
         [HttpGet("{id}")]
-        public string GetComment(int id)
+        public CommentDto GetComment(int id)
         {
-            return $"Comment №{id}";
+            return _commentRepository.GetComment(id);
         }
 
         //  api/comment/by-user/1
         [HttpGet("by-user/{userId}")]
         public List<CommentDto> GetAllCommentsByUserId(int userId)
         {
-            return new List<CommentDto>();
+            return _commentRepository.GetCommentsByUser(userId);
         }
 
         //  api/comment
         [HttpPost]
         public int AddComment([FromBody] CommentAddtInputModel model)
         {
-            return 1;
+            var dto = _mapper.Map<CommentDto>(model);
+            return _commentRepository.AddComment(dto);
         }
 
         //  api/comment/5
         [HttpDelete("{id}")]
         public void DeleteComment(int id)
         {
-
+            _commentRepository.DeleteComment(id);
         }
 
         //  api/comment/5
         [HttpPut("{id}")]
         public string UpdateComment(int id, [FromBody] CommentUpdatetInputModel model)
         {
+            var dto = _mapper.Map<CommentDto>(model);
+            _commentRepository.UpdateComment(id, dto);
```

Comment:
this method is supposed to take only one argument (dto). please change it accordingly